massa-client/src/cmds.rs

```diff
@@ -651,12 +651,24 @@ impl Command {
                         Err(e) => return repl_err!(e),
                     };
 
+                    let mut res = String::new();
+                    if let Ok(addresses_info) = client.public.get_addresses(vec![addr]).await {
+                        for info in addresses_info {
+                            if info.ledger_info.locked_balance > amount {
```

Comment:
Candidate balance is the most relevant here. It's the balance at the most recent blocks that we consider valid. Final balance is the balance at the latest final blocks and locked balance is the balance that is locked after a roll sell.

Moreover, you must check that amount + fee is enough